src/tests/ftest/util/dfuse_utils.py

```diff
@@ -0,0 +1,224 @@
+        super(DfuseCommand, self).__init__("/run/dfuse/*", "dfuse")
+
+        # dfuse options
+        self.puuid = FormattedParameter("-p {}")
+        self.cuuid = FormattedParameter("-c {}")
+        self.mount_dir = FormattedParameter("-m {}")
+        self.svcl = FormattedParameter("-s {}", 0)
+        self.foreground = FormattedParameter("-f", False)
+
+    def get_param_names(self):
+        """Get a sorted list of dfuse command parameter names."""
+        names = self.get_attribute_names(FormattedParameter)
+        return names
+
+    def set_dfuse_params(self, pool, display=True):
+        """Set the dfuse parameters for the DAOS group, pool, and container uuid
+
+        Args:
+            pool (TestPool): DAOS test pool object
+            display (bool, optional): print updated params. Defaults to True.
+        """
+        self.set_dfuse_pool_params(pool, display)
+
+    def set_dfuse_pool_params(self, pool, display=True):
+        """Set Dfuse params based on Daos Pool.
+
+        Args:
+            pool (TestPool): DAOS test pool object
+            display (bool, optional): print updated params. Defaults to True.
+        """
+        self.puuid.update(pool.uuid, "puuid" if display else None)
+        self.set_dfuse_svcl_param(pool, display)
+
+    def set_dfuse_svcl_param(self, pool, display=True):
+        """Set the dfuse svcl param from the ranks of a DAOS pool object.
+
+        Args:
+            pool (TestPool): DAOS test pool object
+            display (bool, optional): print updated params. Defaults to True.
+        """
+
+        svcl = ":".join(
+            [str(item) for item in [
+                int(pool.pool.svc.rl_ranks[index])
+                for index in range(pool.pool.svc.rl_nr)]])
+        self.svcl.update(svcl, "svcl" if display else None)
+
+    def set_dfuse_cont_param(self, cont, display=True):
+        """Set dfuse cont param from Container object
+
+        Args:
+            cont (TestContainer): Daos test container object
+            display (bool, optional): print updated params. Defaults to True.
+        """
+        self.cuuid.update(cont, "cuuid" if display else None)
+
+    def create_dfuse_dir(self, hosts):
+        """Create dfuse directory
+
+        Args:
+            hosts: list of hosts where dfuse directory needs to be created
+
+        Raises:
+            CommandFailure: In case of error creating directory
+        """
+        for host in hosts:
+            if general_utils.check_file_exists(
+                    [host], self.mount_dir.value, directory=True)[0] is False:
+                ret_code = general_utils.pcmd(
+                    [host], "mkdir " + self.mount_dir.value, timeout=30)
+                for key in ret_code:
+                    if key is not 0:
+                        raise CommandFailure(
+                            "DfuseFailure: Error creating directory: "
+                            "{}".format(self.mount_dir.value))
```

Comment:
Report all of the hosts that had issues creating the directory:
```
dir_exists, _ = check_file_exists(
    self.hosts, self.mount_dir.value, directory=True)
if not dir_exists:
    cmd = "mkdir -p {}".format(self.mount_point)
    ret_code = general_utils.pcmd(self.hosts, cmd, timeout=30)
    if 0 not in ret_code:
        error_hosts = NodeSet(
            ",".join(
                [str(node_set) for code, node_set in ret_code.items()
                 if code != 0]))
        raise DfuseFailed(
            "Error creating the {} dfuse mount point on the following "
            "hosts: {}".format(self.mount_point, error_hosts))
```
You could even argue that the check_file_exists() call is not even needed.